example/hooks.ts

```diff
@@ -0,0 +1,158 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from "react";
+
+import {
+  ComputedFeature,
+  LazyLayer,
+  MapRef,
+  SketchEditingFeature,
+  SketchType,
+  ViewerProperty,
+} from "@reearth/core";
+
+import { DEFAULT_CAMERA, DEFAULT_LAYERS, DEFAULT_TILE } from "./constants";
+import { DEFAULT_VIEWER_PROPERTY } from "./scene";
+import { TEST_LAYERS } from "./testLayers";
+import { CESIUM_ION_ACCESS_TOKEN } from "./token";
+
+export default () => {
+  const ref = useRef<MapRef>(null);
+  const [isReady, setIsReady] = useState(false);
+  const handleMount = useCallback(() => {
+    requestAnimationFrame(() => {
+      setIsReady(true);
+    });
+  }, []);
+
+  const handleAPIReady = useCallback(() => {
```

Comment:
We should merge this function with handleMount above if possible.

Reply:
I checked the logic around mount, there's a `ready` props, and only when ready the `children` will be renderered, which means when onMount the Layers Sketch (as children) etc. could be not avaliable sometimes, that's why i expose this independently.